### ISSUE TITLE
Asynchronous commit

### ISSUE THREAD

**Issue by bartoszwalacik** [javers/javers#593]
see https://stackoverflow.com/questions/46700959/async-auditing-with-javers

**Comment by bartoszwalacik** [comment]
released in 3.12.0

**Comment by bartoszwalacik** [comment]
```
    /**
     * Async version of {@link #commit(String, Object)}
     * <br/><br/>
     *
     * <b>Important!</b> Works with MongoDB, not implemented for SQL repositories.
     *
     * @param executor ExecutorService to be used to process commit() asynchronously
     */
    CompletableFuture<Commit> commitAsync(String author, Object currentVersion, Executor executor);
```
